src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -0,0 +1,180 @@
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+
+# Some useful test classes inherited from avocado.Test
+
+from __future__ import print_function
+
+import os
+import json
+
+from avocado import Test as avocadoTest
+from avocado import skip
+
+import fault_config_utils
+import AgentUtils
+import server_utils
+import write_host_file
+from daos_api import DaosContext, DaosLog
+
+# pylint: disable=invalid-name
+def skipForTicket(ticket):
+    ''' Skip a test with a comment about a ticket'''
+    return skip("Skipping until {} is fixed.".format(ticket))
+# pylint: enable=invalid-name
+
+class Test(avocadoTest):
+    '''
+    Basic Test class
+
+    :avocado: recursive
+    '''
+    def __init__(self, *args, **kwargs):
+        super(Test, self).__init__(*args, **kwargs)
+        # set a default timeout of 1 minute
+        # tests that want longer should set a timeout in their .yaml file
+        # all tests should set a timeout and 60 seconds will enforce that
+        if not self.timeout:
+            self.timeout = 60
+
+        item_list = self.logdir.split('/')
+        # pylint: disable=consider-using-enumerate
+        for item_num in range(len(item_list)):
+            if item_list[item_num] == 'job-results':
+                self.job_id = item_list[item_num + 1]
+                break
+        # pylint: enable=consider-using-enumerate
+
+        self.log.info("Job-ID: %s", self.job_id)
+        self.log.info("Test PID: %s", os.getpid())
+
+        self.basepath = None
+        self.orterun = None
+        self.tmp = None
+        self.server_group = None
+        self.daosctl = None
+        self.context = None
+        self.pool = None
+        self.container = None
+        self.hostlist = None
+        self.hostlist_servers = None
+        self.hostfile_servers = None
+        self.d_log = None
+        self.uri_file = None
+        self.fault_file = None
+
+    # pylint: disable=invalid-name
+    def cancelForTicket(self, ticket):
+        ''' Skip a test due to a ticket needing to be completed '''
+        return self.cancel("Skipping until {} is fixed.".format(ticket))
+    # pylint: enable=invalid-name
+
+class TestWithoutServers(Test):
+    '''
+    Run tests without DAOS servers
+
+    :avocado: recursive
+    '''
+    def setUp(self):
+        super(TestWithoutServers, self).setUp()
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as build_vars:
+            build_paths = json.load(build_vars)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+        self.prefix = build_paths['PREFIX']
+        self.tmp = self.prefix + '/tmp'
```

Comment:
Probably also not a major thing, but if making changes anyways, IMO it's worth switching the instances of string concatenation for paths to os.path.join()s, if for nothing else the consistency + never having to worry about leading/trailing slashes in the future.